src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -371,6 +373,11 @@ public void Initialize(ITelemetry telemetry)
             if (string.IsNullOrEmpty(instrumentationKey))
             {
                 instrumentationKey = this.configuration.InstrumentationKey;
+
+                if (string.IsNullOrEmpty(instrumentationKey))
+                {
+                    instrumentationKey = PlatformSingleton.Current.GetEnvironmentVariable(InstrumentationKeyWebSitesEnvironmentVariable);
```

Comment:
Per offline discussion we will change the order so it is: code -> environment variable -> config